prow/plugins.yaml

```diff
@@ -202,7 +202,9 @@ config_updater:
       name: job-config
 
 welcome:
-  message_template: "Welcome @{{.AuthorLogin}}! It looks like this is your first PR to {{.Org}}/{{.Repo}} 🎉"
+- repos:
+  - kubernetes/test-infra
+  message_template: "Welcome @{{.AuthorLogin}}! It looks like this is your first PR to {{.Org}}/{{.Repo}} 🎉🎉"
```

Comment:
IIRC prow will just ignore extra fields, so we can temporarily include the old field here and remove it after the next prow bump?

Reply:
Isn't the problem though that we are changing `plugins.Configuration.Welcome` from `Welcome` to `[]Welcome`?

Reply:
we can test but I think it will just ignore the `-` as an unkown field? IE

```
welcome:
  message_template: "Welcome @{{.AuthorLogin}}! It looks like this is your first PR to {{.Org}}/{{.Repo}} 🎉"
- repos:
  - kubernetes/test-infra
  message_template: "Welcome @{{.AuthorLogin}}! It looks like this is your first PR to {{.Org}}/{{.Repo}} 🎉🎉"
```

Reply:
```
--- FAIL: TestPluginConfig (0.01s)
    welcome_test.go:294: Failed to unmarshal plugin config: error converting YAML to JSON: yaml: line 205: did not find expected key.
```

Reply:
I think the only way to maintain backward compatibility is to add another field `ConfigurableWelcome` (but having a better name), which is of type `[]Welcome` and mark `Welcome` of type `Welcome` as deprecated. And then remove the deprecated type after a certain amount of time.

But this is _very_ ugly.

Also: https://github.com/kubernetes/community/blob/master/contributors/devel/api_changes.md#on-compatibility

Reply:
I sorta like the name, but I'm open to options.

Reply:
> I sorta like the name

I like the name too. :+1: 

Is the welcome plugin used at a lot of other places/prow deployments i.e. will a change here _really_ break a lot of other users?

(not advocating breaking changes, just curious. :smile: )

Reply:
WelcomeNote?
WelcomeMessage?